govc/about/cert.go

```diff
@@ -107,23 +137,45 @@ func (r *certResult) Write(w io.Writer) error {
 func (cmd *cert) Run(ctx context.Context, f *flag.FlagSet) error {
 	u := cmd.URLWithoutPassword()
 	c := soap.NewClient(u, false)
+	t := c.Client.Transport.(*http.Transport)
+	r := certResult{cmd: cmd}
 
 	if err := cmd.SetRootCAs(c); err != nil {
 		return err
 	}
 
-	r := certResult{cmd: cmd}
+	t.DialTLS = func(network string, addr string) (net.Conn, error) {
+		conn, err := tls.Dial(network, addr, t.TLSClientConfig)
+
+		if err == nil {
+			r.state = conn.ConnectionState()
+
+			return conn, nil
+		}
+
+		switch err.(type) {
+		case x509.UnknownAuthorityError:
```

Comment:
Just the errors where we fall through to thumbprint checking, any other error is returned.